From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: 404Setup <153366651+404Setup@users.noreply.github.com>
Date: Fri, 31 May 2024 19:04:49 +0800
Subject: [PATCH] Divine: lithium: entity.fast_elytra_check +
 entity.fast_hand_swing

Original code by CaffeineMC, licensed under LGPL v3
You can find the original code on https://github.com/CaffeineMC/lithium-fabric (Yarn mappings)

diff --git a/src/main/java/net/minecraft/world/entity/LivingEntity.java b/src/main/java/net/minecraft/world/entity/LivingEntity.java
index 3b846f9d5b72f1b3b2a19df3b6a89362245bd3d4..fc4f65d61fa21ed40c561596840e993f03bde648 100644
--- a/src/main/java/net/minecraft/world/entity/LivingEntity.java
+++ b/src/main/java/net/minecraft/world/entity/LivingEntity.java
@@ -2642,6 +2642,7 @@ public abstract class LivingEntity extends Entity implements Attackable {
     }
 
     protected void updateSwingTime() {
+        if (!this.swinging && this.swingTime == 0) return; // DivineMC - lithium: entity.fast_hand_swing
         int i = this.getCurrentSwingDuration();
 
         if (this.swinging) {
@@ -3576,6 +3577,7 @@ public abstract class LivingEntity extends Entity implements Attackable {
     }
 
     private void updateFallFlying() {
+        if (!this.isFallFlying()) return; // DivineMC - lithium: entity.fast_elytra_check
         boolean flag = this.getSharedFlag(7);
 
         if (flag && !this.onGround() && !this.isPassenger() && !this.hasEffect(MobEffects.LEVITATION)) {
